src/ethrpc.ts

```diff
@@ -28,3 +29,25 @@ export const transactionListener = (provider: any, tx: string, userCallback: any
       .catch((err: Error) => userCallback({ error: err.message }))
   , 2000)
 }
+
+export enum PROVIDERS {
+  METAMASK = 'METAMASK',
+  NIFTY = 'NIFTY',
+  WALLET_CONNECT = 'WALLET_CONNECT'
+}
+
+/**
+ * Return the Provider Name. Used to detect DataVault features
+ * @param provider web3Provider
+ */
+export const getProviderName = (provider: any) => {
```

Comment:
Default name? What happens if open the Manager with another wallet?

Reply:
That is a great question and I am open to suggestions. The problem here is that the encryption for the datavault is wallet specific. Some wallets work and others do not, and right now, the only way to know is by us manually testing each wallet to see if it interacts with the encryption and signing for the Datavault. 

Let me research if we can detect if a wallet implements the eth_getEncryptionPublicKey and eth_decrypt methods and base the decision on that. Currently, I'm unsure if this is possible.

Reply:
I think we should assume that if the wallet is not Metamask, we should not try to encrypt. Let's create an issue out of this

Reply:
#48